samtranslator/swagger/swagger.py

```diff
@@ -788,6 +788,50 @@ def add_models(self, models):
 
             self.definitions[model_name.lower()] = schema
 
+    def add_request_parameters_to_method(self, path, method_name, request_parameters):
+        """
+        Add Parameters to Swagger.
+
+        :param string path: Path name
+        :param string method_name: Method name
+        :param list request_parameters: Dictionary of Parameters
+        :return:
+        """
+
+        normalized_method_name = self._normalize_method_name(method_name)
+        # It is possible that the method could have two definitions in a Fn::If block.
+        for method_definition in self.get_method_contents(self.get_path(path)[normalized_method_name]):
+
+            # If no integration given, then we don't need to process this definition (could be AWS::NoValue)
+            if not self.method_definition_has_integration(method_definition):
+                continue
+
+            existing_parameters = method_definition.get('parameters', [])
+
+            for request_parameter in request_parameters:
+
+                parameter_name = request_parameter['Name']
+                location_name = parameter_name.replace('method.request.', '')
+                location, name = location_name.split('.')
+
+                parameter = {
+                    'in': location,
+                    'name': name,
+                    'required': request_parameter['Required'],
+                    'type': 'string'
+                }
+
+                existing_parameters.append(parameter)
+
+                if request_parameter['Caching']:
+
+                    integration = method_definition[self._X_APIGW_INTEGRATION]
+                    cache_parameters = integration.get('cacheKeyParameters', [])
```

Comment:
Can you make `cacheKeyParameters` a constant, similar to `self._X_APIGW_INTEGRATION`?

Reply:
done.